events/s3_object_lambda.go

```diff
@@ -0,0 +1,64 @@
+package events
+
+type S3ObjectLambdaEvent struct {
+	XAmzRequestId        string                `json:"xAmzRequestId"`
+	GetObjectContext     *GetObjectContext     `json:"getObjectContext,omitempty"`
+	ListObjectsContext   *ListObjectsContext   `json:"listObjectsContext,omitempty"`
+	ListObjectsV2Context *ListObjectsV2Context `json:"listObjectsV2Context,omitempty"`
+	HeadObjectContext    *HeadObjectContext    `json:"headObjectContext,omitempty"`
+	Configuration        Configuration         `json:"configuration"`
+	UserRequest          UserRequest           `json:"userRequest"`
+	UserIdentity         UserIdentity          `json:"userIdentity"`
+	ProtocolVersion      string                `json:"protocolVersion"`
+}
+
+type GetObjectContext struct {
+	InputS3Url  string `json:"inputS3Url"`
+	OutputRoute string `json:"outputRoute"`
+	OutputToken string `json:"outputToken"`
+}
+
+type ListObjectsContext struct {
+	InputS3Url string `json:"inputS3Url"`
+}
+
+type ListObjectsV2Context struct {
+	InputS3Url string `json:"inputS3Url"`
+}
+
+type HeadObjectContext struct {
+	InputS3Url string `json:"inputS3Url"`
+}
+
+type Configuration struct {
```

Comment:
All the new structs in the PR should be prefixed with something like `S3ObjectLambda`, to avoid polluting the shared event namespace with very generic names like `Configuration`.

Reply:
Thanks for the suggestion, I added the prefix.